crates/driver/src/extract_defs.rs

```diff
@@ -74,18 +74,16 @@ pub async fn extract_exports(
     let extracted_mappings: ExtractedMappings = read_json_file(&extract.extracted_mappings)?;
 
     let mut work: HashMap<String, Vec<ExtractedMapping>> = HashMap::default();
-    for (k, content_path) in extracted_mappings.relative_path_to_extractmapping.iter() {
-        let directory = if let Some(idx) = k.rfind('/') {
-            let u_k = k.split_at(idx).0;
-
-            u_k.rfind('/').map(|e| u_k.split_at(e).0).unwrap_or(u_k)
+    for (rel_path, content_path) in extracted_mappings.relative_path_to_extractmapping.iter() {
+        let entry = if !opt.no_aggregate_source {
+            to_directory(&rel_path)
         } else {
-            k.as_str()
+            rel_path.to_string()
         };
-        if let Some(v) = work.get_mut(directory) {
+        if let Some(v) = work.get_mut(&entry) {
```

Comment:
nit: this entry to directory rename could be in a separate PR to make this diff smaller and easier to review.

Reply:
Here's a PR to split out some of the directory/entry renames - https://github.com/bazeltools/bzl-gen-build/pull/122